src/meltano/cli/add.py

```diff
@@ -19,13 +22,33 @@
 from meltano.core.plugin_install_service import PluginInstallReason
 from meltano.core.project_add_service import ProjectAddService
 from meltano.core.tracking.contexts import CliEvent, PluginsTrackingContext
+from meltano.core.yaml import yaml
 
 if t.TYPE_CHECKING:
     from meltano.core.plugin.project_plugin import ProjectPlugin
     from meltano.core.project import Project
     from meltano.core.tracking import Tracker
 
 
+def _load_yaml_from_ref(_ctx, _param, value: str | None) -> dict:
+    if not value:
+        return
+
+    try:
+        url = urlparse(value)
+        if url.scheme and url.netloc:
+            response = requests.get(value, timeout=10)
+            response.raise_for_status()
+            content = response.text
+        else:
+            content = Path(value).read_text()
```

Comment:
This means that the option accepts two types of args:

1. URLs, e.g. `https://example.com/path/to/plugin.yaml`
2. File paths (either absolute or relative), e.g. `/Users/path/to/plugin.yaml`

right?

Reply:
Correct. Unsure if "ref" as terminology covers this, but that is the intended behaviour.

Reply:
- File path is intended for local development of taps to test with `meltano add`
- URL is intended for Hub PR checks (i.e. `--from-ref https://github.com/meltano/hub/blob/main/_data/meltano/extractors/tap-spotify/matatika.yml` - see https://github.com/meltano/hub/issues/1191), and maybe helps move towards a model where the plugin definition lives in the source repository - similar to how dbt Hub works IIRC (if that's something you guys are interested in)